### PR TITLE
Add 'where' mask argument to random.categorical.

### PR DESCRIPTION
**Pull request by carlosgmartin**
Adds a `where` mask argument to [random.categorical](https://docs.jax.dev/en/latest/_autosummary/jax.random.categorical.html).

It is common in RL (action masking) and other domains to want to exclude certain elements from being sampled. Adding a `where` argument to the function provides a user-friendly, convenient way to accomplish this.

### CODE REVIEW

**Comment by jakevdp** [comment]
Thanks for looking at this. To me, passing `logits=jnp.where(where, p, -jnp.inf)` directly seems much clearer than requiring a new argument in the API. A downside of the approach here is that in general it makes the shape of the result more difficult to reason about (I suppose it would be something like `np.delete(np.broadcast_shapes(p.shape, where.shape), axis)`?). Given this, I think it's better not to add the new argument. What do you think?
